src/main/java/in/zapr/druid/druidry/client/DruidJerseyClient.java

```diff
@@ -0,0 +1,131 @@
+package in.zapr.druid.druidry.client;
+
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.glassfish.jersey.apache.connector.ApacheClientProperties;
+import org.glassfish.jersey.apache.connector.ApacheConnectorProvider;
+import org.glassfish.jersey.client.ClientConfig;
+
+import java.util.List;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.GenericType;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import in.zapr.druid.druidry.client.exception.ConnectionException;
+import in.zapr.druid.druidry.client.exception.QueryException;
+import in.zapr.druid.druidry.query.DruidQuery;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class DruidJerseyClient implements DruidClient {
+
+    private static final int DEFAULT_CONNECTION_POOL_LIMIT = 5;
+    private final String druidUrl;
+    private final DruidConfiguration druidConfiguration;
+
+    private ClientConfig jerseyConfig;
+    private Client client;
+    private WebTarget queryWebTarget;
+
+    public DruidJerseyClient(@NonNull DruidConfiguration druidConfiguration,
+                             ClientConfig jerseyConfig) {
+
+        this.druidUrl = druidConfiguration.getUrl();
+        this.jerseyConfig = jerseyConfig;
+
+        log.debug("Will query to {}", druidUrl);
+        this.druidConfiguration = druidConfiguration;
+    }
+
+    @Override
+    public void connect() throws ConnectionException {
+
+        PoolingHttpClientConnectionManager connectionManager
+                = new PoolingHttpClientConnectionManager();
+
+        int numberOfConnectionsInPool = DEFAULT_CONNECTION_POOL_LIMIT;
+
+        if (this.druidConfiguration.getConcurrentConnectionsRequired() != null) {
+            numberOfConnectionsInPool = this.druidConfiguration.getConcurrentConnectionsRequired();
+        }
+
+        connectionManager.setDefaultMaxPerRoute(numberOfConnectionsInPool);
+
+        try {
+            if (jerseyConfig == null) {
+                this.jerseyConfig = new ClientConfig();
+            }
+
+            this.jerseyConfig.property(ApacheClientProperties.CONNECTION_MANAGER, connectionManager);
+            this.jerseyConfig.connectorProvider(new ApacheConnectorProvider());
+
+            this.client = ClientBuilder.newClient(this.jerseyConfig);
+            this.queryWebTarget = this.client.target(this.druidUrl);
+
+        } catch (Exception e) {
+            throw new ConnectionException(e);
+        }
+    }
+
+    @Override
+    public void close() throws ConnectionException {
+        try {
+            this.client.close();
```

Comment:
It is better to add null check on client before calling close() as we throw ConnectionException in connect() method in case of any exception